File: java/src/array/MaxOccurredAlphabet.java
package array;

import java.util.ArrayList;
import java.util.HashMap;

public class MaxOccurredAlphabet {
    public static void main(String[] args) {
        // 알파벳 빈도수 제일 큰 값 찾기
//        String str = "hello my name is mason";
        String str = "we love algorithm";
        frequency(str);
    }

    static void frequency(String str) {
        HashMap<Character, Integer> map = new HashMap<>();

        for (char c : str.toCharArray()) {
            if (!Character.isWhitespace(c)) {
                map.put(c, map.getOrDefault(c, 0) + 1);
            }
        }

        // 최대 빈도 구하기
        int max = 0;
        for (int i : map.values()) {
            if (max < i) {
                max = i;
            }
        }

        ArrayList<String> list = new ArrayList<>();
        for (Character c : map.keySet()) {
            if (map.get(c) == max) {
                list.add(String.valueOf(c));
            }
        }

        System.out.println(String.join(", ", list));
    }
}

File: python/array/max_occurred_alphabet.py
def max_occurred(str):
    counter = {}

    for i in str:
        if i != ' ':
            if i in counter:
                counter[i] += 1
            else:
                counter[i] = 1

    max_val = max(counter.values())
    result = [k for k, v in counter.items() if v == max_val]

    return ', '.join(result)


if __name__ == "__main__":
    print("정답 : m | 풀이 : ", max_occurred("hello my name is mason"))
    print("정답 : e, l, o | 풀이 : ", max_occurred("we love algorithm"))
    print("정답 : b, e ,t | 풀이 : ", max_occurred("best of best youtube"))
